frontend/ui/menusorter.lua

```diff
@@ -22,11 +21,10 @@ local MenuSorter = {
 function MenuSorter:readMSSettings(config_prefix)
     if config_prefix then
         local menu_order = string.format(
-            "%s/%s_menu_order", DataStorage:getSettingsDir(), config_prefix)
+            "%s/%s_menu_order.lua", DataStorage:getSettingsDir(), config_prefix)
 
-        if lfs.attributes(menu_order..".lua") then
-            return require(menu_order) or {}
-        end
+        local ok, data = pcall(dofile, menu_order)
```

Comment:
I prefer the original (not talking about the require). Okay, so the user made a typo… but that's the point! They should be immediately notified that they messed up, shouldn't they?

Reply:
Concretely, this is conflating "file exists" and "file has syntax error".

I think you can alternatively write it along these lines:
```lua
if not ok and not data:match(menu_order) then
    -- Syntax error, it'll contain the filename in a file not found error
    -- `cannot open bla.lua: No such file or directory`
    error("immediate proper feedback")
end